pact/Pact/Core/Serialise/LegacyPact/Types.hs

```diff
@@ -0,0 +1,1821 @@
+      JD.withObject "KeySetName" $ \o -> KeySetName
+        <$> o JD..: "ksn"
+        <*> (fromMaybe Nothing <$> o JD..:? "ns")
+
+-- data Meta = Meta
+--   { _mDocs  :: !(Maybe Text) -- ^ docs
+--   , _mModel :: ![Exp Info]   -- ^ models
+--   } deriving (Eq, Show, Generic)
+
+-- instance JD.FromJSON Meta where parseJSON = lensyParseJSON 2
+
+-- | Pact syntax expressions
+-- data Exp i =
+--   ELiteral (LiteralExp i) |
+--   EAtom (AtomExp i) |
+--   EList (ListExp i) |
+--   ESeparator (SeparatorExp i)
+--   deriving (Eq,Ord,Generic,Functor,Foldable,Traversable,Show)
+
+-- instance JD.FromJSON i => JD.FromJSON (Exp i) where
+--   parseJSON v =
+--     (ELiteral <$> parseJSON v) <|>
+--     (EAtom <$> parseJSON v) <|>
+--     (EList <$> parseJSON v) <|>
+--     (ESeparator <$> parseJSON v)
+
+-- data LiteralExp i = LiteralExp
+--   { _litLiteral :: !Literal
+--   , _litInfo :: !i
+--   } deriving (Eq,Ord,Generic,Functor,Foldable,Traversable,Show)
+
+
+-- instance JD.FromJSON i => JD.FromJSON (LiteralExp i) where
+--   parseJSON = JD.withObject "LiteralExp" $ \o ->
+--     LiteralExp <$> o JD..: "lit" <*> o JD..: expInfoField
+
+
+-- data AtomExp i = AtomExp
+--   { _atomAtom :: !Text
+--   , _atomQualifiers :: ![Text]
+--   , _atomDynamic :: Bool
+--   , _atomInfo :: i
+--   } deriving (Eq,Ord,Generic,Functor,Foldable,Traversable,Show)
+
+-- instance JD.FromJSON i => JD.FromJSON (AtomExp i) where
+--   parseJSON = JD.withObject "AtomExp" $ \o ->
+--     AtomExp <$> o JD..: "atom" <*> o JD..: "q"
+--     <*> (fromMaybe False <$> o JD..:? "dyn")
+--     <*> o JD..: expInfoField
+
+-- data ListExp i = ListExp
+--   { _listList :: ![Exp i]
+--   , _listDelimiter :: !ListDelimiter
+--   , _listInfo :: !i
+--   } deriving (Eq,Ord,Generic,Functor,Foldable,Traversable,Show)
+
+-- instance JD.FromJSON i => JD.FromJSON (ListExp i) where
+--   parseJSON = JD.withObject "ListExp" $ \o ->
+--     ListExp <$> o JD..: "list" <*> o JD..: "d" <*> o JD..: expInfoField
+
+-- data ListDelimiter = Parens|Brackets|Braces deriving (Eq,Show,Ord,Generic,Bounded,Enum)
+
+
+-- instance JD.FromJSON ListDelimiter where
+--   parseJSON = withText "ListDelimiter" $ \t -> case t of
+--     "()" -> pure Parens
+--     "[]" -> pure Brackets
+--     "{}" -> pure Braces
+--     _ -> fail "Invalid ListDelimiter"
+
+-- data SeparatorExp i = SeparatorExp
+--   { _sepSeparator :: !Separator
+--   , _sepInfo :: !i
+--   } deriving (Eq,Ord,Generic,Functor,Foldable,Traversable,Show)
+
+
+-- instance JD.FromJSON i => JD.FromJSON (SeparatorExp i) where
+--   parseJSON = JD.withObject "SeparatorExp" $ \o ->
+--     SeparatorExp <$> o JD..: "sep" <*> oJD..: expInfoField
+
+-- data Separator = Colon|ColonEquals|Comma deriving (Eq,Ord,Generic,Bounded,Enum,Show)
+
+
+-- instance JD.FromJSON Separator where
+--   parseJSON = withText "Separator" $ \t -> case t of
+--     ":" -> pure Colon
+--     ":=" -> pure ColonEquals
+--     "," -> pure Comma
+--     _ -> fail "Invalid separator"
```

Comment:
Lots of commented instances. Are these not needed?

Reply:
Cleaned them up